foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/security/JCEEncryptor.java

```diff
@@ -125,10 +127,15 @@ private static SecretKey getAESGCMMultitasker() throws Exception {
         private static byte[] getIvGCM() {
             byte[] ivGCM = new byte[IV_GCM_LENGTH];
             SecureRandom random = null;
-            try {
-                random = SecureRandom.getInstanceStrong();
-            } catch (NoSuchAlgorithmException e) {
-                throw new RuntimeException(e);
+            String useStrongRNG = PrivilegedAccessHelper.getSystemProperty(SystemProperties.SECURITY_ENCRYPTOR_USE_STRONG_RANDOM_NUMBER_GENERATOR);
+            if (Boolean.parseBoolean(useStrongRNG)) {
```

Comment:
It will change default behavior into less secure mode by default.
But I'd like rephrase it and change condition content. If users needs it e.g. due performance reasons, they could enable it by system property like
`SECURITY_ENCRYPTOR_USE_NON_BLOCKING_RANDOM_NUMBER_GENERATOR = "eclipselink.security.encryptor.non.blocking.random.number.generator";` -> `random = new SecureRandom();`
For me in this case security  is higher, than performance so I'd like keep as a default `random = SecureRandom.getInstanceStrong();` and conditionally `random = new SecureRandom();`

Reply:
I understand your concerns.
But IMHO, this pull request does not immediately lower the security level, because it does not change the cipher (AES GCM) for encryption, and using `new SecureRandom()` as random source is still secure.

From the viewpoint of availability, the impact of stall (almost hang in some environments) due to blocking algorithms cannot be ignored.
If a blocking algorithm is used by default, users might encounter the stall in a production environment, which can cause business losses.
There are several related discussions discouraging use of `SecureRandom.getInstanceStrong()` in server-side applications:

- https://www.blackduck.com/blog/proper-use-of-javas-securerandom.html
- https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/general-recommendations-for-secure-coding/#java-specific-recommendations_13
- https://github.com/akka/akka/issues/20908

Should we still use `SecureRandom.getInstanceStrong()`?

Reply:
OK I accepted Your arguments. I don't like blocked production systems too. Lets use `new SecureRandom` as a default.